### PR TITLE
update to api v3

### PR DESCRIPTION
**Pull request by axolile**

### CODE REVIEW

File: MDS/Collivery/Model/MdsCollivery.php [hunk 1]
```diff
@@ -42,116 +46,151 @@ public function __construct(array $config = [], $cache = null)
...
-            if (! $this->init()) {
-                return false;
-            }
+            $query = http_build_query($data);
+            $client = curl_init($url.'?'.$query);
+        }
 
-            $user_email    = $this->config->user_email;
-            $user_password = $this->config->user_password;
+        curl_setopt($client, CURLOPT_SSL_VERIFYPEER, false);
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/MdsCollivery.php]: Do not turn off SSL verification.

File: MDS/Collivery/Observer/Shipment.php [hunk 2]
```diff
@@ -128,11 +128,11 @@ public function execute(Observer $observer)
                     'cust_ref' => $order->getRealOrderId()
                 ];
 
-                $validatedCollivery = $this->validateCollivery($validateData);
-                is_null($validatedCollivery) && $this->errorBag($this->getErrors());
+                /*$validatedCollivery = $this->validateCollivery($validateData);
```
Comment (WilRyan) [inline MDS/Collivery/Observer/Shipment.php]: Please remove commented out code

**Review by WilRyan — APPROVED** [review]

File: MDS/Collivery/Model/MdsCollivery.php [hunk 3]
```diff
@@ -42,116 +46,151 @@ public function __construct(array $config = [], $cache = null)
...
-                if ($this->check_cache != 0) {
-                    $this->cache->put('collivery.auth', $authenticate, 50);
-                }
+        $headerArray = [
+            'X-App-Name:'.$this->config->app_name.' mds/collivery/class',
+            'X-App-Version:'.$this->config->app_version,
+            'X-App-Host:'.$this->config->app_host,
+            'X-App-Url'     => $this->config->app_url,
+            'X-App-Lang:'.'PHP '.phpversion(),
+            'Content-Type: application/json'
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/MdsCollivery.php]: Please add `Accept: application/json`

File: MDS/Collivery/Model/MdsCollivery.php [hunk 4]
```diff
@@ -168,26 +207,32 @@ public function getTowns($country = "ZAF", $province = null)
...
+            if (isset($result['data'])) {
                 if (is_null($province)) {
                     if ($this->check_cache != 0) {
-                        $this->cache->put('collivery.towns.' . $country, $result['towns'], 60*24);
+                        $this->cache->put('collivery.towns.' . $country, $result['data'], 60*24);
                     }
                 } else {
                     if ($this->check_cache != 0) {
-                        $this->cache->put('collivery.towns.' . $country . '.' . $province, $result['towns'], 60*24);
+                        $this->cache->put('collivery.towns.' . $country . '.' . $province, $result['dat'], 60*24);
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/MdsCollivery.php]: Check typo on `$result['dat']`

File: MDS/Collivery/Model/MdsCollivery.php [hunk 5]
```diff
@@ -286,20 +337,24 @@ public function getLocationTypes()
             return $this->cache->get('collivery.location_types');
         } else {
             try {
-                $result = $this->client()->get_location_types($this->token);
-            } catch (SoapFault $e) {
-                $this->catchSoapFault($e);
+                $param =["api_token" => ""];
+                if($this->token!=null)
+                    $param["api_token"]=$this->token;
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/MdsCollivery.php]: Always add `{ .. }` to `if` (and all) statements 
Please see https://www.php-fig.org/psr/psr-1/ and https://www.php-fig.org/psr/psr-12/

File: MDS/Collivery/Observer/Shipment.php (hunk 2 above)
Comment (MarnuLombard) [inline MDS/Collivery/Observer/Shipment.php]: Remove the commented out code

File: MDS/Collivery/Model/Carrier/Collivery.php [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 namespace MDS\Collivery\Model\Carrier;
 
+use Klarna\Core\Model\Api\Rest\Service;
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/Carrier/Collivery.php]: Please remove this import, it's not used.

**Review by MarnuLombard — CHANGES_REQUESTED** [review]

File: MDS/Collivery/Model/MdsCollivery.php [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <?php
 namespace MDS\Collivery\Model;
 
-use SoapClient; // Use PHP Soap Client
+use Exception;
 use SoapFault;  // Use PHP Soap Fault
```
Comment (MarnuLombard) [inline MDS/Collivery/Model/MdsCollivery.php]: Why is the `SoapFault` being used if there's no SOAP in this class?

**Review by MarnuLombard — APPROVED** [review]

File: MDS/Collivery/Model/MdsCollivery.php (hunk 7 above)
Comment (axolile) [reply MDS/Collivery/Model/MdsCollivery.php]: Removed all refs to SOAP

**Pull request merged**